### DISCUSSION TITLE
Accelerometer, sensor update frequency

### DISCUSSION THREAD

**Discussion by SashaCX75** [zepp-health/zeppos-forum#282] [Developer Feedback]
I wanted to check the update frequency of the accelerometer sensor and found a strange behaviour.
One time the data refresh time is about 500 milliseconds.
Then one time the data refresh time is about 12 milliseconds.
Then several times the data refresh time is about 4 milliseconds.
Then the cycle repeats. That is, the time between sensor actuations is 500-12-4-4-4-4-4-4-4-4-500-12-12....
That is, the spread between the fastest and longest sensor response is more than 100 times. 

How normal is such sensor operation? 
A delay of 500 milliseconds I think is a lot. It makes it impossible to use the sensor in interactive applications. It turns out that the data is updated only 2 times per second and it makes the movement based on the accelerometer sensor very jerky.

I tried to change the refresh rate with setFreqMode(FREQ_MODE_HIGH), but it didn't give any results.

**Comment by silver-zepp** [comment]
Which watch model do you experience this behavior on? We will investigate this and update you as soon as possible. @SashaCX75

**Comment by SashaCX75** [comment]
@silver-zepp Amazfit Balance

It seems like the accelerometer sensor is only available on ZeppOS 3, and that system is only on one model so far. Or maybe I missed something.

**Comment by MyoungXUE** [comment]
@SashaCX75 Thank you for your feedback, I will check the frequency problem of the sensor.
